owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/StatusRequestor.kt

```diff
@@ -0,0 +1,113 @@
+            setConnectionTimeout(TRY_CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+        }
+    }
+
+    data class RequestResult(
+        val getMethod: GetMethod,
+        val status: Int,
+        val result: RemoteOperationResult<OwnCloudVersion>,
+        val redirectedToUnsecureLocation: Boolean
+    )
+
+    fun requestAndFollowRedirects(baseLocation: String, client: OwnCloudClient): RequestResult {
+        var currentLocation = baseLocation
+        var redirectedToUnsecureLocation = false
+        var status: Int
+
+        while (true) {
+            val getMethod = getGetMethod(currentLocation)
+
+            status = client.executeHttpMethod(getMethod)
+            val result =
+                if (isSuccess(status)) RemoteOperationResult<OwnCloudVersion>(RemoteOperationResult.ResultCode.OK)
+                else RemoteOperationResult(getMethod)
+
+            if (result.redirectedLocation.isNullOrEmpty() || result.isSuccess) {
+                return RequestResult(getMethod, status, result, redirectedToUnsecureLocation)
+            } else {
+                val nextLocation = updateLocationWithRedirectPath(currentLocation, result.redirectedLocation)
+                redirectedToUnsecureLocation =
+                    checkIfConnectionIsRedirectedToNoneSecure(
+                        redirectedToUnsecureLocation,
+                        currentLocation,
+                        nextLocation
+                    )
+                currentLocation = nextLocation
+            }
+        }
+    }
+
+    private fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK
+
+    fun handleRequestResult(
+        requestResult: RequestResult,
+        baseUrl: String
+    ): RemoteOperationResult<OwnCloudVersion> {
+        if (!isSuccess(requestResult.status))
+            return RemoteOperationResult(requestResult.getMethod)
+
+        val respJSON = JSONObject(requestResult.getMethod.getResponseBodyAsString() ?: "")
+        if (!respJSON.getBoolean(NODE_INSTALLED))
+            return RemoteOperationResult(RemoteOperationResult.ResultCode.INSTANCE_NOT_CONFIGURED)
+
+        val ocVersion = OwnCloudVersion(respJSON.getString(NODE_VERSION))
+        // the version object will be returned even if the version is invalid, no error code;
+        // every app will decide how to act if (ocVersion.isVersionValid() == false)
+        val result =
+            if (requestResult.redirectedToUnsecureLocation) {
+                RemoteOperationResult<OwnCloudVersion>(RemoteOperationResult.ResultCode.OK_REDIRECT_TO_NON_SECURE_CONNECTION)
+            } else {
+                if (baseUrl.startsWith(HTTPS_SCHEME)) RemoteOperationResult(
+                    RemoteOperationResult.ResultCode.OK_SSL
+                )
+                else RemoteOperationResult(RemoteOperationResult.ResultCode.OK_NO_SSL)
+            }
+        result.data = ocVersion
+        return result
+    }
+}
```

Comment:
Add a new line at the end of file 👍  Same for test file

Reply:
done